omero_marshal/encode/encoders/well.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Glencoe Software, Inc. All rights reserved.
+#
+# This software is distributed under the terms described by the LICENCE file
+# you can find at the root of the distribution bundle.
+# If the file is missing please request a copy by contacting
+# jason@glencoesoftware.com.
+#
+
+from ... import SCHEMA_VERSION
+from .annotation import AnnotatableEncoder
+from omero.model import WellI
+
+
+class Well201501Encoder(AnnotatableEncoder):
+
+    TYPE = 'http://www.openmicroscopy.org/Schemas/OME/2015-01#Well'
+
+    def encode(self, obj):
+        v = super(Well201501Encoder, self).encode(obj)
+        self.set_if_not_none(v, 'Column', obj.column)
+        self.set_if_not_none(v, 'Row', obj.row)
+        self.set_if_not_none(v, 'ExternalDescription', obj.externalDescription)
+        self.set_if_not_none(v, 'ExternalIdentifier', obj.externalIdentifier)
+        self.set_if_not_none(v, 'Type', obj.type)
+        # FIXME: Color conversion required (uses integer encoded Color field)
+        self.set_if_not_none(v, 'Alpha', obj.alpha)
+        self.set_if_not_none(v, 'Red', obj.red)
+        self.set_if_not_none(v, 'Green', obj.green)
+        self.set_if_not_none(v, 'Blue', obj.blue)
+        self.set_if_not_none(v, 'omero:status', obj.status)
+
+        if obj.isWellSamplesLoaded() and obj.sizeOfWellSamples() > 0:
+            wellsamples = list()
+            for wellsample in obj.copyWellSamples():
```

Comment:
When ```wellsample is None``` above, I see that ```obj.copyWellSamples()``` gives 2 wellsamples, the first of which is None.
The other can be seen in the DB:
```
omero=# select * from wellsample where well=2717;
  id  | permissions | posx | posy | timepoint | version | creation_id | external_id | group_id | owner_id | update_id | image | plateacquisition | well | well_index | posxunit | posyunit 
------+-------------+------+------+-----------+---------+-------------+-------------+----------+----------+-----------+-------+------------------+------+------------+----------+----------
 2653 |         -40 |      |      |           |         |      635023 |             |        5 |        3 |    635023 | 15084 |             1567 | 2717 |          1 |          | 
(1 row)
```
The Image name for that WellSample is "ticket-10564_6x4y1z1t1c8b-swatch-2s2p4w2ws.ome [6x6x1x8-swatch.tif-1]" which suggests that this is quite a contrived example.
When I look at the ```plate-2217``` in webclient, I see that it has 4 fields in the drop-down select. In field-1, the plate has 4 wells with images, but in all the other fields, only 1 of those 4 wells has an image.
Clicking on any of those 4 Wells, with any field fails to load right panel with
```